src/main/groovy/com/moowork/gradle/node/npm/NpmCiTask.groovy

```diff
@@ -0,0 +1,27 @@
+package com.moowork.gradle.node.npm
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.tasks.TaskAction
+
+/**
+ * Task for making npmInstall use npm ci instead of npm install.*/
+class NpmCiTask
+    extends DefaultTask
+{
+    public final static String NAME = 'npmCi'
+
+    NpmCiTask()
+    {
+        this.group = 'Node'
+        this.description = 'Make npm installation use the "ci" command'
+        this.outputs.upToDateWhen { false }
+    }
+
+    @TaskAction
+    void configure() {
+        project.tasks.withType(NpmInstallTask.class).each {
+            it.npmCommand = ['ci']
+            project.logger.debug("Reconfiguring: ${it.name}" )
+        }
+    }
```

Comment:
The solution will probably work, but I did not actually test it. IMHO the current solution is not very straightforward and appears to be a little brittle. The re-configuration is some magic that one would not expect and which is likely to break. The tasks have bi-directional dependencies:
* `NpmCiTask` knows implementation details of `NpmInstallTask`. And for expected behavior it also DEPENDS on NpmInstallTask, otherwise it would be a no op (if sticking to this solution we should probably add the `dependsOn` task dependency).
* `NpmInstallTask` must run after `NpmCiTask`.

My proposed solution would be option 1 as outlined in issue #7 . This would mean:
* Provide `NpmCiTask` as a more or less copy & paste version of `NpmInstallTask`.
* Remove the task dependencies.
* Provide a configuration option to set the appropriate task dependencies. Values for this option could be "npmCi", "npmInstall", and "none". The default should be "npmInstall" to keep backwards compatibility.

Do you see any drawbacks with this solution?